lib/query/query_supporters.rb

```diff
@@ -199,16 +199,16 @@ def self.full_filter_expr(np_id, query)
       .order_by('payments.max_date DESC NULLS LAST')
 
     if query[:last_payment_after].present?
-      expr = expr.and_where("payments.max_date > $d", d: Chronic.parse(query[:last_payment_after]))
+      expr = expr.and_where("payments.max_date > timezone(COALESCE(nonprofits.timezone, \'UTC\'), timezone(\'UTC\', $d))", d: Chronic.parse(query[:last_payment_after]).beginning_of_day)
```

Comment:
I added this `.beginning_of_day` because I realized that the problem that was happening initially with the payments page (the one that is described on the issue) didn't happen at the supporters page:

![image](https://user-images.githubusercontent.com/15739610/119183135-0c33ce00-ba4a-11eb-91e1-d6204a80be9d.png)

And then I found out that it was searching for dates before or after xx/yy at 12:00:00 UTC. This is merely to force it to start at 00:00:00.

Reply:
Do you know why it was searching for dates based on noon UTC?

Reply:
I couldn't find where it was being set, my guess was that the time picker chose that time, but I'm not sure.

Reply:
Hmmm, looking at the network traffic, it is sending the date by itself, no adjustment for time of day. That's so weird.